From: Anoob Soman <anoob.soman@citrix.com>
Date: Wed, 19 Jul 2017 10:59:48 +0100
Subject: [PATCH 1/3] xen-platform: Handle write of four-byte build number to
 io-port 0x10

XEN PV driver writes four-byte buildnumber to io-port 0x10, store it in
PCIXenPlatformState

Signed-off-by: Anoob Soman <anoob.soman@citrix.com>
---
 hw/i386/xen/xen_platform.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hw/i386/xen/xen_platform.c b/hw/i386/xen/xen_platform.c
index 4004d55..5f2c489 100644
--- a/hw/i386/xen/xen_platform.c
+++ b/hw/i386/xen/xen_platform.c
@@ -63,6 +63,7 @@ typedef struct PCIXenPlatformState {
     uint8_t flags; /* used only for version_id == 2 */
     int drivers_blacklisted;
     uint16_t driver_product_version;
+    uint32_t driver_build_number;
 
     /* Log from guest drivers */
     char log_buffer[4096];
@@ -219,9 +220,12 @@ static void platform_fixed_ioport_writew(void *opaque, uint32_t addr, uint32_t v
 static void platform_fixed_ioport_writel(void *opaque, uint32_t addr,
                                          uint32_t val)
 {
+    PCIXenPlatformState *s = opaque;
+
     switch (addr) {
     case 0:
         /* PV driver version */
+        s->driver_build_number = val;
         break;
     }
 }
